### ISSUE TITLE
TPUEstimator does not update meta-weights

### ISSUE THREAD

**Issue by KirillAI** [tensorflow/tensorflow#45173]
**System information**

- OS Platform and Distribution: Debian GNU/Linux 9.11 (stretch)
- TensorFlow version: 2.1.0
- Python version: 3.5.3

I train a neural network using `TPUEstimator`. In `model_fn`, losses are stored in `losses` list. After the end of an epoch, meta-weights `theta` should be updated using `losses`. But for the entire epoch, only one tensor is added to `losses` (`len(losses) = 1`) and `theta` remains the same (`theta==theta_prev = True`).

Here is [gist](https://gist.github.com/KirillAI/ed88ac43b3da1643629b08e76bd0db5c) with the complete code. Part of `model_fn`:

```
...

    def update_theta():
        global losses
        if len(losses) > 1:
            losses_by_updates = [[] for _ in range(num_updates)]
            for i in range(num_tasks):
                for j in range(num_updates):
                    losses_by_updates[j].append(losses[i*num_updates+j])
            losses_by_updates = [tf.reduce_sum(loss) / tf.to_float(num_tasks) for loss in losses_by_updates]
            gradients = theta_optimizer.compute_gradients(losses_by_updates[-1])
            with tf.variable_scope(tf.get_variable_scope(), reuse=tf.AUTO_REUSE):
                theta_op = theta_optimizer.apply_gradients(gradients)
            weights = copy_theta_to_weights()
            return theta_op
        else:
            return tf.no_op()
...

    loss = 0.0
    for key in logits_train_dic:
        logit_train = logits_train_dic[key]
        loss += tf.square(labels[key]-logit_train)
    loss_op = tf.reduce_mean(loss)
    losses.append(loss_op)

...
```

Is there any way to write loss to a list inside `model_fn`? Or maybe there is some other way to update the meta-weights using `TPUEstimator`?

**Comment by Saduf2019** [comment]
@KirillAI 
Could you please let us know if this is still an issue in latest stable TF v2.6.0 ?Thank you!
